### DISCUSSION TITLE
Unable to set OpenAI thru .env

### DISCUSSION THREAD

**Discussion by krishnan2704** [microsoft/semantic-kernel#7360] [General]
Using the step as below :
### Configuring the Kernel

Let's get started with the necessary configuration to run Semantic Kernel. For Notebooks, we require a `.env` file with the proper settings for the model you use. Create a new file named `.env` and place it in this directory. Copy the contents of the `.env.example` file from this directory and paste it into the `.env` file that you just created.

**NOTE: Please make sure to include `GLOBAL_LLM_SERVICE` set to either OpenAI, AzureOpenAI, or HuggingFace in your .env file. If this setting is not included, the Service will default to AzureOpenAI.**

I am trying to set up OpenAI, however the flow is defaulting to AzureOpenAI.   I am using Visual Studio.  Anyone help me to resolve?

**Comment by moonbox3** [comment]
Hi @krishnan2704, are you encountering this issue while trying to run the getting_started notebooks in Python? or while trying to run concept samples?

**Reply by krishnan2704** [reply]
Yes I am trying to run samples in getting_started notebook in python.  I am using Visual Studio Code,  I have followed the instruction and created .env file.  I am using OpenAI, however the env is not recognized and default to AzureOpenAI
![setup](https://github.com/user-attachments/assets/32741cc0-2b96-494a-ab29-c66de30df1d2)

**Comment by moonbox3** [comment]
Hi @krishnan2704, thanks for the details. I found a bug in the service settings config. It should be `settings = ServiceSettings.create()` instead of `settings = ServiceSettings()`. As a temp fix, until we release a new package early next week, I'd recommend making this update (to include `.create()`) on your end, or hard code the service as needed.

[Here](https://github.com/microsoft/semantic-kernel/pull/7366) is a link to the PR with the fix.

**Comment by krishnan2704** [comment]
Thank you so much it works!!
Just noticed that service_id is set from env, whereas to use functions like OpenAIChatCompletion, I need to pass model and apiKey explicitly while calling (not taking from .env).

**Reply by moonbox3** [reply]
Hi @krishnan2704, if you do have your `OPENAI_API_KEY = ""` configured in the .env file, it will first take it from there. If you explicitly pass in the `api_key` to the `OpenAIChatCompletion` constructor, that will take precedence. `service_id` is never handled from the .env file, and should always be explicitly passed in to a constructor, where applicable. Otherwise, if not passed in, the value `default` for service_id is used.

**Comment by moonbox3** [comment]
Fixed as of #7366
